pkg/auth/auth.go

```diff
@@ -0,0 +1,173 @@
+package auth
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strings"
+
+	logging "github.com/op/go-logging"
+)
+
+// ConfigEntry - Configuration for authentication
+type ConfigEntry struct {
```

Comment:
Think the other packages take the pattern of naming this `Config` for each section, so it's referenced as `Config` within the package, and `auth.Config` outside.